kernel-spark/src/main/java/io/delta/kernel/spark/read/SparkMicroBatchStream.java

```diff
@@ -64,6 +64,9 @@ public class SparkMicroBatchStream implements MicroBatchStream, SupportsAdmissio
   private final boolean shouldValidateOffsets;
   private final SparkSession spark;
 
+  // Tracks whether this is the first batch for this stream (no checkpointed offset).
```

Comment:
can you write down the assumptions around this. that this boolean is used with the assumption that the following sequence of methods will be called - initialOffset -> latestOffset -> ... and then set to false.